sherpa/instrument.py

```diff
@@ -540,122 +570,104 @@ def __call__(self, model, session=None):
 
-            if size is None:
-                kargs['size'] = kshape
-                # update size param to default
-                self.size = kargs['size']
+            if self.size is None:
+                kwargs['size'] = kshape
 
         else:
             if (self.kernel is None) or (not callable(self.kernel)):
                 raise PSFErr('nopsf', self._name)
+
             kshape = data.get_dims()
-            # (kargs['lo'], kargs['hi'],
-            # kargs['width']) = _get_axis_info(kargs['args'], dshape)
 
-            kargs['lo'] = [1] * len(kshape)
-            kargs['hi'] = kshape
-            kargs['width'] = [1] * len(kshape)
+            kwargs['lo'] = [1] * len(kshape)
+            kwargs['hi'] = kshape
+            kwargs['width'] = [1] * len(kshape)
 
-            if center is None:
-                kargs['center'] = [int(dim / 2.) for dim in dshape]
-                # update center param to default
-                self.center = kargs['center']
+            if self.center is None:
+                kwargs['center'] = [int(dim / 2.) for dim in dshape]
 
-            if size is None:
-                kargs['size'] = dshape
-                # update size param to default
-                self.size = kargs['size']
+            if self.size is None:
+                kwargs['size'] = dshape
 
-            kargs['is_model'] = True
+            kwargs['is_model'] = True
             if hasattr(self.kernel, 'pars'):
                 # freeze all PSF model parameters if not already.
                 for par in self.kernel.pars:
                     par.freeze()
 
             if hasattr(self.kernel, 'thawedpars'):
-                kargs['frozen'] = (len(self.kernel.thawedpars) == 0)
+                kwargs['frozen'] = (len(self.kernel.thawedpars) == 0)
+
+        if self.center is None:
+            # update center param to default
+            self.center = kwargs['center']
+
+        if self.size is None:
+            # update size param to default
+            self.size = kwargs['size']
 
-        is_kernel = (kargs['is_model'] and not kargs['norm'] and
+        is_kernel = (kwargs['is_model'] and not kwargs['norm'] and
                      len(kshape) == 1)
+
         # Handle noticed regions for convolution
         if numpy.iterable(data.mask):
-            kargs['do_pad'] = True
-            kargs['pad_mask'] = data.mask
+            kwargs['do_pad'] = True
+            kwargs['pad_mask'] = data.mask
 
         if is_kernel:
-            for id in ['is_model', 'lo', 'hi', 'width', 'size']:
-                kargs.pop(id)
-            self.model = Kernel(dshape, kshape, **kargs)
+            for kwarg in ['is_model', 'lo', 'hi', 'width', 'size']:
+                kwargs.pop(kwarg)
+
+            self.model = Kernel(dshape, kshape, **kwargs)
             return
 
-        if radial:
-            self.model = RadialProfileKernel(dshape, kshape, **kargs)
+        if int(self.radial.val):
```

Comment:
I suspected that in practice, this is only used with positive numbers (dimension, number of pixels) and that this guards against 0.000000001, which should be True to be rounded down to 0 because of machine-precision. But this probably needs a comment, give that you don't remember either...

Reply:
Aha: this is from the original code, which has

```python
(size, center, origin, kargs['norm'], radial) = (self.size, self.center, self.origin, bool_cast(self.norm.val), int(self.radial.val))
```

I have added a comment i #1477 to say `# TODO: why is this not 'self.radial.val > 0'?` but I don't want to change the code here.